tensorflow/python/keras/wrappers/scikit_learn.py

```diff
@@ -114,7 +114,7 @@ def get_params(self, **params):  # pylint: disable=unused-argument
     Returns:
         Dictionary of parameter names mapped to their values.
     """
-    res = copy.deepcopy(self.sk_params)
+    res = self.sk_params.copy()
```

Comment:
Just curious about the issue here. The deepcopy will create all new instances for any nested param in the self.sk_params, whereas the self.sk_params.copy() could return same instance for nested param here. Isn't the deepcopy the intended behavior (isolated the original instance and copied version)?

Reply:
The original issue was filing in the keras issue, so it got lost in the switch of development over to the tensorflow issue.

The summary is:  we actually want a reference here, not a copy.  So `deep copy` is wrong.

More detail here: https://github.com/keras-team/keras/issues/13586

Reply:
I see. It seems that more like a sklearn problem since it uses "is" to compare, rather than "==". I guess this PR is just to walkaround that issue (otherwise the deepcopy should be the intended behavior). Could u please add a comment above this line, and put the reference issue here?

Reply:
The usage of ``is`` instead of ``==`` is intentional, we want to guarantee that parameters are not modified in ``__init__``. Constructor parameters can be arbitrarily complex and so checking for equality might be not feasible or nor appropriate.